src/Symfony/Bridge/Doctrine/Validator/Constraints/UniqueEntityValidator.php

```diff
@@ -111,7 +111,27 @@ public function validate($entity, Constraint $constraint)
             }
         }
 
-        $repository = $em->getRepository(get_class($entity));
+        if (null !== $constraint->repository) {
+            /* Retrieve repository from given entity name.
+             * We ensure the retrieved repository can handle the entity
+             * by checking the entity is the same, or subclass of the supported entity.
+             */
+            $repository = $em->getRepository($constraint->repository);
+            $supportedClass = $repository->getClassName();
+
+            $className = $class->getName();
+
+            if (!($supportedClass === $className || is_subclass_of($className, $supportedClass))) {
+                throw new ConstraintDefinitionException(sprintf(
+                    "Unable to use the given '%s' repository for the '%s' entity.",
```

Comment:
I must have done the same as [here](https://github.com/ogizanagi/symfony/blob/doctrinebridge/repo_option/src/Symfony/Bridge/Doctrine/Validator/Constraints/UniqueEntityValidator.php#L85) I guess. 
There is some other occurrences of "bad" quoting messages in Symfony components then.

Anyway, fixed. :)